File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        diff = {} # key : diff , value :index
        for idx,num in enumerate(nums): #enumerate函數可以遍歷所有數值
            if diff.get(num,None)==None: #整數不在目標內
                diff[target-num] =idx
            else:
                return [diff[num] , idx]

File: 1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int ,int> index;
        for(int i=0;i<nums.size();i++)
        {
            index[nums[i]]=i;
        } 
        for(int i=0;i<nums.size();i++)
        {
            int left = target - nums[i];
            if(index.count(left) && index[left] !=i ) //查找map中是否有其元素
            {
                return {i,index[left]};
            }
        }
        return {};
    }
};